core/environment/transition_configure.go

```diff
@@ -173,22 +179,33 @@ func (t ConfigureTransition) do(env *Environment) (err error) {
 		for {
 			log.Debug("waiting for workflow to become active")
 			select {
-			case wfStatus = <-notify:
+			case wfStatus = <-notifyStatus:
 				log.WithField("status", wfStatus.String()).
 				    Debug("workflow status change")
 				if wfStatus == task.ACTIVE {
 					break WORKFLOW_ACTIVE_LOOP
 				}
 				continue
 			case <-time.After(deploymentTimeout):
-				err = errors.New("workflow deployment timed out")
+				err = errors.New(fmt.Sprintf("workflow deployment timed out. timeout: %s",deploymentTimeout.String()))
 				break WORKFLOW_ACTIVE_LOOP
+			// This is needed for when the workflow fails during the STAGING state(mesos status),mesos responds with the `REASON_COMMAND_EXECUTOR_FAILED`,
+			// By listening to workflow state ERROR we can break the loop before reaching the timeout (1m30s), we can trigger the cleanup faster
+			// in the CreateEnvironment (environment/manager.go) and the lock in the `envman` is reserved for a sorter period, which allows operations like
+			// `environment list` to be done almost immediatelly after mesos informs with TASK_FAILED.
+			case wfState := <-notifyState:
+				if wfState == task.ERROR {
+					log.WithField("state", wfState.String()).
+				    	Debug("workflow state change")
+					err = errors.New("workflow deployment failed")
```

Comment:
Can we also right in this spot query for **which** task(s) failed?
I expect that the WF state `ERROR` will be reported when the first `taskRole` (and therefore the first task) goes to `ERROR`. Can we query & print here which task (or role) it was that failed?

Reply:
I need to check if it is possible to retrieve the task or the `taskid`.